test.py

```diff
@@ -1,87 +1,122 @@
-from sys import argv
-from pathlib import Path
-from os import walk, system
-from ast import literal_eval
 import subprocess
+from ast import literal_eval
+from dataclasses import dataclass
+from enum import Enum
+from os import system
+from pathlib import Path
+from sys import argv
+
+
+class ExpectedOutputType(Enum):
+    """Possible test results"""
+
+    FAIL = 1
+    EXIT_WITH_CODE = 2
+    EXIT_WITH_OUTPUT = 3
+
+
+@dataclass(frozen=True)
+class Expected:
+    """A container for the expected output of a test"""
+
+    expected_output_type: ExpectedOutputType
+    value: str | int | None
+
 
-def get_expected(filename):
-    expected = {}
-    with open(filename) as f:
-        for line in f:
+def get_expected(filename) -> Expected | None:
+    expected = None
+    with open(filename) as file:
+        for line in file:
+            # exit as soon as the first non-commented line is found
             if not line.startswith("///"):
                 break
+
+            # remove the comment marker and any leading or trailing
+            # whitespace from the line
             line = line[3:].strip()
+
+            # if the line does not contain any characters after the
+            # comment marker, continue
```

Comment:
Everything else looks good, but these comments don't really seem to add any value.

Reply:
2dd0ae4